### ISSUE TITLE
[BUG] Inline images cannot open more than once

### ISSUE THREAD

**Issue by darius07** [laurencedawson/sync-for-reddit#968]
**Describe the bug**
Inline images open correctly only once. The image cache must be manually cleared in order for the inline image to open again. Peaking on inline images works correctly multiple times, and inline gifs can be opened multiple times with no issues.

**To Reproduce**
1. Open a post containing inline images
2. Click on an inline image
3. Close the inline image
4. Click again the same inline image
5. Open the debug bottomsheet
6. Inline image is stuck at: **Submitting to the ImageManager**

**Expected behavior**
Inline images open correctly multiple times, without manually clearing the image cache after each inline image opening.

**Screenshot and Video**
Example post: https://www.reddit.com/r/redditsync/comments/tmpmfr/v22324_beta_release/

The inline image is stuck at Submitting to the ImageManager
![Screenshot_20220331-110050](https://user-images.githubusercontent.com/12880926/161007184-6cf40a2f-7ee6-4109-b8ef-1a8de405792f.png)

Video example: https://i.imgur.com/sW3idQG.mp4

**Smartphone:**
 - Device: Lenovo Phab2
 - OS: Android 6
 - Version v22.3.28

**Comment by laurencedawson** [comment]
Very strange, I've not had anyone else report this one at all.

Are you still having this issue?

**Comment by darius07** [comment]
Yes, still having the same issue.

While troubleshooting this, I encountered another issue also solved by manually clearing the image cache. Tall images do not open (peaking works), or only open once. I have to clear the image cache, and the image can open successfully only once. 

https://user-images.githubusercontent.com/12880926/161820933-45c54b56-c0c8-446c-8fa0-173725942dce.mp4

**Comment by darius07** [comment]
I left the beta, and tried sync pro v19 again, and the issue is not present there. I can open inline and tall images more than once, without problems, and without manually clearing the image cache.

Rejoined the beta, and the issue came back on v22.3.28. I also have tried deleting the app data / cache from Android's app menu, reinstalling the app, trying login out and in, trying the Anonymous sync account, but nothing makes a difference.

**Comment by laurencedawson** [comment]
Still trying to replicate this!

**Comment by darius07** [comment]
On the new V22.4.20 beta, I retested the issue, and even the resolution and size is not fetched properly when the images fails to load.

Inline image loading issue:
Example inline image comment: 
https://www.reddit.com/r/redditsync/comments/u72362/v22420_beta_and_v19020_release/i5bwrlh/
1) Image cache manually cleared
2) Inline image opens normally
3) Upon closing and re-opening the inline image, it is still stuck at Submitting to the ImageManager
![Screenshot_20220419-185640](https://user-images.githubusercontent.com/12880926/164046187-e4a704b7-0ef8-433e-ae39-46a065087319.png)
![Screenshot_20220419-185656](https://user-images.githubusercontent.com/12880926/164046241-27336df8-d485-4f11-973a-3b753d4b7c6c.png)


Tall image loading issue:
Example tall image post: 
https://www.reddit.com/r/polandball/comments/u6zfnn/fables_but_its_ruined_by_countries
1) Image cache manually cleared
2) Tall image opens normally
3) Upon closing and re-opening the tall image, it is again stuck at Submitting to the ImageManager
![Screenshot_20220419-185439](https://user-images.githubusercontent.com/12880926/164046084-a03e0144-3c9e-4fbd-9ed8-2e7893f0b63f.png)
![Screenshot_20220419-185505](https://user-images.githubusercontent.com/12880926/164046136-13be6bf5-a216-4133-a7de-811a7d4f1d96.png)

**Comment by laurencedawson** [comment]
I've added an additional error check for the next release. Fingers crosssed...

**Comment by darius07** [comment]
No luck even on version v22.4.23-18. I have no clue what is causing it, but it seems linked to image caching, and reading from the image cache. Very weird that normal sized images do not suffer from it, only tall/inline images break and load infinitely.

I also tried this: 
- Opened the image in Sync Free v19.0.20 → Share image (Link) → Opens successfully first time in Sync Pro v22.4.23-18. 
- Shared the same image a second time from Sync Free v19.0.20 → Sync Pro v22.4.23-18, image loads infinitely.

![Screenshot_20220424-185203a](https://user-images.githubusercontent.com/12880926/164985838-67b2d207-9655-4557-8af0-0987886add5f.png)

**Comment by laurencedawson** [comment]
Interesting. Thanks for the update.

**Comment by laurencedawson** [comment]
Follow up. Still having the issue?

Yet to get another report of this. Strange one.

**Comment by darius07** [comment]
The issue still persists. It is very strange indeed.

Inline GIFs and image albums have no problem opening whatsoever once or 100 times, but a single inline image either cannot open at all (if cached) or only opens once (after it is cached).

Tall images behave similarly, if I open them just after they load, when scrolling down to load more posts, they open. The rest of the time I have to use image peek, if I want to see the tall image at all. Alternatively, I have to manually clear the image cache, but that still allows me to open any tall image only once.

**Comment by laurencedawson** [comment]
Next thing to try.

Can you disable this:

Settings > Animations  > Image transitions

**Comment by darius07** [comment]
No difference. 

I also tried with all the toggles inside Animations disabled, tried different on/off toggles in Media, tried switching between the three options for Image Preloading (advanced) in Data Options, but still does not solve the issue.

**Comment by laurencedawson** [comment]
Are you based in the UK?

**Comment by darius07** [comment]
Not based in UK, but in Europe. I have joined the Sync for Reddit Discord channel.

**Comment by darius07** [comment]
It seems the issue is related to Android 6 and/or Lenovo Phab2's software. I do not have the issue any more on Android 12.
